MatrixSDK/Space/MXSpaceNotificationCounter.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  MXSpaceNotificationCounter.swift
+//  MatrixSDK
+//
+//  Created by Gil Eluard on 19/09/2021.
+//
+
+import Foundation
+
+@objcMembers
+public class MXSpaceNotificationState: NSObject {
+    public var favouriteMissedDiscussionsCount: UInt = 0
+    public var favouriteMissedDiscussionsHighlightedCount: UInt = 0
+    public var directMissedDiscussionsCount: UInt = 0
+    public var directMissedDiscussionsHighlightedCount: UInt = 0
+    public var groupMissedDiscussionsCount: UInt = 0
+    public var groupMissedDiscussionsHighlightedCount: UInt = 0
+    public var allCount: UInt {
+        return favouriteMissedDiscussionsCount + directMissedDiscussionsCount + groupMissedDiscussionsCount
+    }
+    public var allHighlightCount: UInt {
+        return favouriteMissedDiscussionsHighlightedCount + directMissedDiscussionsHighlightedCount + groupMissedDiscussionsHighlightedCount
+    }
+    
+    static func +(left: MXSpaceNotificationState, right: MXSpaceNotificationState) -> MXSpaceNotificationState {
+        let sum = MXSpaceNotificationState()
+        sum.favouriteMissedDiscussionsCount = left.favouriteMissedDiscussionsCount + right.favouriteMissedDiscussionsCount
+        sum.favouriteMissedDiscussionsHighlightedCount = left.favouriteMissedDiscussionsHighlightedCount + right.favouriteMissedDiscussionsHighlightedCount
+        sum.directMissedDiscussionsCount = left.directMissedDiscussionsCount + right.directMissedDiscussionsCount
+        sum.directMissedDiscussionsHighlightedCount = left.directMissedDiscussionsHighlightedCount + right.directMissedDiscussionsHighlightedCount
+        sum.groupMissedDiscussionsCount = left.groupMissedDiscussionsCount + right.groupMissedDiscussionsCount
+        sum.groupMissedDiscussionsHighlightedCount = left.groupMissedDiscussionsHighlightedCount + right.groupMissedDiscussionsHighlightedCount
+        return sum
+    }
+    
+    static public func +=( left: inout MXSpaceNotificationState, right: MXSpaceNotificationState) {
+        left = left + right
+    }
+}
+
+@objcMembers
+public class MXSpaceNotificationCounter: NSObject {
+    
+    // MARK: - Properties
+    
+    public var homeNotificationState = MXSpaceNotificationState()
```

Comment:
Is this property can be modified from the outside? Otherwise can make the setter private:
```suggestion
    public private(set) var homeNotificationState = MXSpaceNotificationState()
```